qiskit/primitives/decorators/allow.py

```diff
@@ -388,3 +369,8 @@ def wrapper(
     setattr(cls, "__call__", wrapper)
     setattr(cls, "__init__", init_wrapper)
     return cls
+
+
+def _finditer(obj: object, object_ids: list[int]) -> Iterator[int]:
```

Comment:
How about making a function with `next(_finditer(...))`? All `_finditer` is used with `next`.

Reply:
Thank you. I thought it would be easier to read since `next` here could fail and the error would be caught by try-except.